api/shared-modules/src/node/utils/encryption/index.ts

```diff
@@ -50,20 +49,16 @@ export const decrypt = (cipher: string, secret: string) => {
 	return decrpyted.toString();
 };
 
-export const asymEncrypt = (data: any, privateKey: string, channelPublicKey: string): string => {
+export const asymEncrypt = (data: any, privateKey: string, publicKey: string): string => {
 	const diffie = crypto.createDiffieHellman(bs58.decode(privateKey));
-	const sharedKey = diffie.computeSecret(bs58.decode(channelPublicKey))
-	const dataBytes = aesjs.utils.utf8.toBytes(JSON.stringify(data));
-	const aesCtr = new aesjs.ModeOfOperation.ctr(sharedKey);
-	const encrypted = aesCtr.encrypt(dataBytes);
-	return bs58.encode(encrypted)
+	const sharedKey = diffie.computeSecret(bs58.decode(publicKey))
+	const encrypted = encrypt(JSON.stringify(data), bs58.encode(sharedKey))
+	return encrypted
 }
 
-export const asymDecrypt = (encrypted: string, privateKey: string, channelPublicKey: string): string => {
+export const asymDecrypt = (encrypted: string, privateKey: string, publicKey: string): string => {
 	const diffie = crypto.createDiffieHellman(bs58.decode(privateKey));
-	const sharedKey = diffie.computeSecret(bs58.decode(channelPublicKey))
-	const aesCtr = new aesjs.ModeOfOperation.ctr(sharedKey);
-	const decoded = bs58.decode(encrypted);
-	const decrypted = aesCtr.decrypt(decoded)
-	return JSON.parse(aesjs.utils.utf8.fromBytes(decrypted))
+	const sharedKey = diffie.computeSecret(bs58.decode(publicKey))
+	const decrypted = decrypt(encrypted, bs58.encode(sharedKey))
+	return JSON.parse(decrypted)
```

Comment:
I don't think you should call JSON.parse here since we expect a string.

Reply:
true